app/models/compose_validator.rb

```diff
@@ -100,16 +100,16 @@ def validate_expose_format(value, path, errors)
       end
 
       def validate_service_name(value, path, errors)
-        regex = /^[a-zA-Z0-9]*$/
-        regex_with_build = /^[a-z0-9]*$/
+        regex = /^[a-zA-Z0-9]+(?:[._-][a-zA-Z0-9]+)*$/
+        regex_with_build = /^[a-z0-9]+(?:[._-][a-z0-9]+)*$/
         service_name = path[0]
         if value.include?('build')
           unless service_name =~ regex_with_build
-            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-z0-9]', path)
+            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-z0-9._-] but not starting with [.-_]', path)
           end
         else
           unless service_name =~ regex
-            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-zA-Z0-9]', path)
+            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-zA-Z0-9._-] but not starting with [.-_]', path)
```

Comment:
@alexwelch For a service name that has a 'build' component to it, docker-compose throws this error:
```
Building _custom_code-ver.1...
repository name component must match "[a-z0-9]+(?:[._-][a-z0-9]+)*"
```
But for a service name that does not use 'build', I noticed that docker-compose did not throw a validation error, but it did choke on the name, and the service would not start. See the output below:
```
Creating app2_custom_code-ver.1_1...
Creating app2__Web_service-ver.2_1...
Attaching to app2_custom_code-ver.1_1, app2__Web_service-ver.2_1
^CGracefully stopping... (press Ctrl+C again to force)
Stopping app2__Web_service-ver.2_1... done
```
So, I decided to disallow a service name to start with [.-_] to be consistent and not be buggy. 

What are your thoughts?